topics/hashing/easy/valid-hashcodes.tex

```diff
@@ -14,6 +14,16 @@
 \end{lstlisting}
 
 \begin{parts}
+\part What does it mean for a hashcode to be valid?
+\begin{solution}[0.75in]
+The general contract of hashCode is:
+\begin{itemize}
+\item Whenever it is invoked on the same object more than once during an execution of a Java application, the hashCode method must consistently return the same integer, provided no information used in equals comparisons on the object is modified. This integer need not remain consistent from one execution of an application to another execution of the same application.
```

Comment:
Maybe we should simplify this? It might be confusing for students to think about the execution of a Java application. Instead, we can just say that if hashcode is invoked on the same object multiple times, it must consistently return the same integer if nothing about the object changes.